File: Java/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit, buy;
        profit = 0;
        buy = Integer.MAX_VALUE;        
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < buy)
                buy = prices[i];            
            if (prices[i] - buy > profit)
                profit = prices[i] - buy;            
        }
        return profit;
    }
}

File: Python/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if prices is None or len(prices) == 0:
            return 0
        buy = prices[0]
        profit = 0
        for p in prices:
            profit = max(p - buy, profit)
            if p < buy:
                buy = p
        return profit